src/main/java/com/rocketden/main/service/ProblemService.java

```diff
@@ -61,9 +67,30 @@ public ProblemDto getProblem(int id) {
         return problems;
     }
 
-    public ProblemTestCaseDto createTestCase(int problemId, CreateTestCaseRequest request) {
-        Problem problem = repository.findById(problemId)
-                .orElseThrow(() -> new ApiException(ProblemError.NOT_FOUND));
+    public ProblemDto getRandomProblem(ProblemSettingsDto request) {
+        ProblemDifficulty difficulty = request.getDifficulty();
+        List<Problem> problems;
+
+        if (difficulty == null || difficulty == ProblemDifficulty.RANDOM) {
```

Comment:
In what situations could a difficulty not be specified? If it's not supposed to be allowed, should it return an error instead of assuming a random difficulty?

Reply:
My assumption behind it was that if no difficulty is specified, that means that the user doesn't care and is open to getting anything, hence treating it the same as random - lmk if you think I should change it tho and I can do that in a separate PR

Reply:
Curious on @helenli8181's thoughts, but I think my preference would skew towards making the user specify this attribute? But like this is the smallest of the teeny-tiny nits, so either way I don't think you have to worry too much.

Reply:
Yeah sure that works - just pushed a commit to update this

Reply:
I'll merge this for now (eager to move past it haha) but if @helenli8181 you feel differently lmk and I can update it in the next PR